graph/src/firehose/endpoints.rs

```diff
@@ -140,15 +140,38 @@ impl FirehoseEndpoint {
 
         debug!(logger, "Retrieving block(s) from firehose");
 
-        let mut block_in_longuest_chain: Option<BlockPtr> = None;
+        let mut latest_received_block: Option<BlockPtr> = None;
         while let Some(message) = block_stream.next().await {
```

Comment:
We can improve this in the future, but I think this `while` loop might be able to be turned into a `loop` statement, since it returns a value on `break` 🙂

Reply:
Not familiar with this construct, do you have an example somewhere?

Reply:
I mean this: https://doc.rust-lang.org/reference/expressions/loop-expr.html#break-and-loop-values